helper/resource/testing.go

```diff
@@ -640,6 +642,10 @@ type TestStep struct {
 	// Custom state checks can be created by implementing the [statecheck.StateCheck] interface, or by using a StateCheck implementation from the provided [statecheck] package.
 	ConfigStateChecks []statecheck.StateCheck
 
+	// ConfigQueryResultChecks allow assertions to be made against the query file during a Config test using a query check.
```

Comment:
nit: Similar note to the other comment about package docs + naming. If we are inspecting a "query file" then i think the naming all makes sense (similar to inspecting a plan file or state file).

If we are inspecting logs from a query command, perhaps a different name would be better suited